pkg/image/registryclient/client.go

```diff
@@ -223,7 +248,51 @@ func (c *Context) Repository(ctx context.Context, registry *url.URL, repoName st
 	if limiter == nil {
 		limiter = rate.NewLimiter(rate.Limit(5), 5)
 	}
-	return NewLimitedRetryRepository(repo, c.Retries, limiter), nil
+	retryRepo := NewLimitedRetryRepository(repo, c.Retries, limiter)
+	if c.AlternateRepositoryStrategy != nil {
+		retryRepo.alternateStrategy = c.AlternateRepositoryStrategy
+	}
+	return retryRepo, nil
+}
+
+// RepositoryWithAlternateReference returns the next cached repository and ref, or will gather alternate sources then return first repo, ref, or error
+// Once the repository is returned it is removed from the cache.
+func (c *Context) RepositoryWithAlternateReference(ctx context.Context, locator imagereference.DockerImageReference, cachedRepos map[imagereference.DockerImageReference]distribution.Repository) (distribution.Repository, imagereference.DockerImageReference, error) {
```

Comment:
no, that would be used in a different strategy than the one I've provided.

Reply:
in `oc` it will be used 1 time, to get the first valid Repository. To get multiple valid Repositories, caller would repeat. Again, I was under the assumption there was a need for this. It's not required by `oc` and not required by image-registry.

Reply:
I've updated. ptal